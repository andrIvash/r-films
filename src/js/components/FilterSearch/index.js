// @flow
import React, { Component } from 'react';

type Props = {
 changeFilter: (elem: HTMLElement) => void,
 selected: string
}

export default class FilterSearch extends Component<Props> {

  _changeFilter = event => this.props.changeFilter(event.target.value);

  render() {
    const { selected } = this.props;
    return (
      <div className='filter-search'>
        <div className='filter-search__title'>Search By</div>
        <div className='btn-group-toggle'>
          <label
            className={selected === 'genres' ?
            'filter-search__label btn btn-secondary active' :
            'filter-search__label btn btn-secondary'}
            htmlFor='genre'
            >
            <input
              autoComplete='off'
              className='filter-search__btn'
              id='genre'
              name='options'
              onChange={this._changeFilter}
              type='radio'
              value='genres'
            /> Genre
          </label>
          <label
            className={selected === 'title' ?
            'filter-search__label btn btn-secondary active' :
            'filter-search__label btn btn-secondary'}
            htmlFor='title'
            >
            <input
              autoComplete='off'
              className='filter-search__btn'
              id='title'
              name='options'
              onChange={this._changeFilter}
              type='radio'
              value='title'
            /> Title
          </label>
        </div>
      </div>
    );
  }
}

