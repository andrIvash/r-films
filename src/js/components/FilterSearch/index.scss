.filter-search {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.filter-search__title {
  font-family: $sign-font;
  font-size: rem(12px);
  color: $color-layout;
  text-transform: uppercase;
  margin-right: rem(10px);
}

.filter-search__label {
  width: rem(70px);
  height: rem(30px);
  font-family: $sign-font;
  font-size: rem(10px);
  margin: 0 rem(5px);
  text-transform: uppercase;
}