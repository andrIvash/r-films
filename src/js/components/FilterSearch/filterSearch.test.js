import React from 'react';
import FilterSearch from './index';

let props = {};

describe('FilterSearch', () => {
  beforeEach(() => {
    props = {
<<<<<<< HEAD
      selected: 'genres',
      changeFilter: jest.fn(),
=======
      onChange: jest.fn(),
      selected: 'genres',
>>>>>>> source
    };
  });
  it('should be defined', () => {
    expect(FilterSearch).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = render(
      <FilterSearch {...props} /> );
    expect(wrapper).toMatchSnapshot();
  });

  it(`should calls onChange 
    when genres input is selected`, () => {
    const wrapper = shallow(
      <FilterSearch {...props} /> );
    wrapper.find('#genre').simulate('change', { target: {value: 'genre' }});
    expect(props.changeFilter).toHaveBeenCalledTimes(1);
    expect(wrapper.find('#genre').parent().hasClass('active')).toBe(true);
  });

  it(`should calls onChange 
    when title input is selected`, () => {
    props.selected = 'title';

    const wrapper = shallow(
      <FilterSearch {...props} /> );
    wrapper.find('#title').simulate('change', { target: {value: 'title' }});
    expect(props.changeFilter).toHaveBeenCalledTimes(1);
    expect(wrapper.find('#title').parent().hasClass('active')).toBe(true);
  });
});
