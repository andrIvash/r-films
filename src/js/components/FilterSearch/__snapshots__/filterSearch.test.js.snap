// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSearch should render correctly 1`] = `
<div
  className="filter-search"
>
  <div
    className="filter-search__title"
  >
    Search By
  </div>
  <div
    className="btn-group-toggle"
  >
    <label
      className="filter-search__label btn btn-secondary"
      htmlFor="genre"
    >
      <input
        autoComplete="off"
        className="filter-search__btn"
        id="genre"
        name="options"
        type="radio"
        value="genres"
      />
       Genre
    </label>
    <label
      className="filter-search__label btn btn-secondary"
      htmlFor="title"
    >
      <input
        autoComplete="off"
        className="filter-search__btn"
        id="title"
        name="options"
        type="radio"
        value="title"
      />
       Title
    </label>
  </div>
</div>
`;
